app/lib/features/events/widgets/events_item.dart

```diff
@@ -35,7 +48,10 @@ class EventItem extends ConsumerWidget {
               ),
             ),
             const SizedBox(width: 10),
-            _buildRsvpStatus(context, ref),
+            Visibility(
+              visible: isShowRsvp,
+              child: _buildRsvpStatus(context, ref),
+            ),
```

Comment:
There is a strong want in this team to use `Visibility` a lot more often than needed. And I'd like to quote the flutter docs about it again:
> Using this widget is not necessary to hide children. The simplest way to hide a child is just to not include it, or, if a child must be given (e.g. because the parent is a [StatelessWidget](https://api.flutter.dev/flutter/widgets/StatelessWidget-class.html)) then to use [SizedBox.shrink](https://api.flutter.dev/flutter/widgets/SizedBox/SizedBox.shrink.html) instead of the child that would otherwise be included. 

In this case, we do not do any animation or need to transition that item ever. So why go through the extra render at all?

Instead this should really just be:
```suggestion
            if (isShowRsvp) _buildRsvpStatus(context, ref),
```

Reply:
That might not be necessary as a approach to cover all conditional render cases but I see in a lot of places, we are checking a lot of cases wrapped up in one condition that certainly doesn't look good in widget tree. In that ```Visibility``` widget makes it easier to understand what purpose it's serving. Although I agree, that for this simplistic case, its more easier to read it in one line ;)

Reply:
Thanks @gnunicorn for highlight this things. Will take care of it.